### PR TITLE
[stable/memcached] Add more memcached options and optional Prometheus metrics

### PR DESCRIPTION
**Pull request by whereisaaron**
**What this PR does / why we need it**:

Hi @gtaylor, thank you for this chart, I extended it:
1. To be able to set some more important `memcached` options, like `--max-item-size`
2. To have the option to deploy with Prometheus `memcached-exporter` metrics sidecar

**Special notes for your reviewer**:

More detailed changes:
* Bump `memcached` to 1.5.9 (including in `Chart.yaml`)
* Bump chart version to 2.3.1 in `Chart.yaml`
* Add support for `memcached` options `--port`, `--udp-port`, `--max-item-size`, `--conn-limit`, `--slab-min-size`, `--slab-growth-factor`
* Update `README.md` and `values.yaml` for new options
* Use use appropriate StatefulSet `apiVersion` for k8s version
* Add optional Prometheus `memcached-exporter` sidecar container (disabled by default)
* Update StatefulSet to use parallel rollout/delete as `memcached` replicas have no order dependence
* Update StatefulSet to use `RollingUpdate` strategy for changes (was defaulting to `OnDelete` when using old `apps/v1beta1` API version, so `helm upgrade` didn't use to work)
* Update resource requests in `values.yaml` to include limits but be commented out by default

When the review is complete, I will squash and rebase this PR.

### CODE REVIEW

**Comment by whereisaaron** [comment]
@paulczar @mattfarina would you mind reviewing this PR? It is an earlier PR from July that I believe is a superset of everything in the more recent #6847 you reviewed. In addition to adding a Prometheus metrics sidecar like #6847, it adds some of the main `memcached` configuration options available in recent versions. All configuration options are included in the updated `READMe.md` and `NOTES.txt`.

**Comment by olemarkus** [comment]
Thanks for the PR. I just became a maintainer of this chart and will go through all the open issues/PRs soon. 

Generally I don't want too many changes in one single PR though. All those extra args could also be solved with a more generic extraArgs flag. Only the most important ones need to be configurable directly. That way the chart also becomes more future proof.

File: stable/memcached/Chart.yaml [hunk 1]
```diff
@@ -1,11 +1,13 @@
 name: memcached
-version: 2.2.0
-appVersion: 1.5.6
+version: 2.3.2
```
Comment (mattfarina) [inline stable/memcached/Chart.yaml:2]: Why did the chart version go up to `2.3.2` instead of `2.3.0`?

File: stable/memcached/README.md [hunk 2]
```diff
@@ -40,17 +46,41 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `memcached.slabMinSizeBytes`       | Minimum slab size                         | `48`                                                    |
+| `memcached.slabGrowthFactor`       | Slab growth factor                        | `1.25`                                                  |
+| `memcached.verbosity`              | Verbosity level (v, vv, or vvv)           | `v`                                                     |
+| `memcached.extendedOptions`        | Comma-separated options                   | `modern`                                                |
+| `metrics.enabled`                  | Enable Prometheus metrics exporter        | `false`                                                 |
+| `metrics.images.repository`        | Metrics exporter repository               | `prom/memcached-exporter`                               |
+| `metrics.images.tag`               | Metrics exporter image tag                | A recent version compatible with memcached version      |
+| `metrics.images.pullPolicy`        | Metrics exporter image pull policy        | `IfNotPresent`                                          |
```
Comment (mattfarina) [inline stable/memcached/README.md:71]: Why is the metrics image pull policy different from the memcached image pull policy?

**Comment by olemarkus** [comment]
I started to review this PR, but honestly it is hard with so many changes going on. Would you mind splitting? I should be able to provide swift approvals then.

**Comment by olemarkus** [comment]
Kenneth and I decided to pull in #6847 pending the review of this PR. The content of this one is still very much interesting though.

File: stable/memcached/Chart.yaml (hunk 1 above)
Comment (whereisaaron) [reply stable/memcached/Chart.yaml:2]: Thanks @mattfarina, we use this chart in production, and while it has been waiting for review it has gone through several rounds of internal improvements, so the version has been bumped several times in our repository. The original PR only bumped it 2.2.0 to 2.2.1 to add some settings, but we've done a lot of additional work on this that we wanted to try to contribute back.

I assume if anything gets merged we'll update this in the PR based on whatever the current master version is.

File: stable/memcached/README.md (hunk 2 above)
Comment (whereisaaron) [reply stable/memcached/README.md:71]: In Pods each container has its own pull policy setting. I don't a view that they shouldn't be tied to the same value in the chart, but its seemed cleaner to have uniform repository/tag/pullPolicy setting for each container. And it doesn't prevent users setting both to the same value.

**Pull request closed without merging**